### PR TITLE
Required

### PR DESCRIPTION
**Pull request by mbouzi**
make all fields in partner form required
- located at "/partner-form"
- user cannot continue without putting a value or choice for step
- all user input values get sent to customer io

### CODE REVIEW

File: layouts/partner-form/single.html [hunk 1]
```diff
@@ -43,6 +47,27 @@
       }
     }
 
+		// if vale is input or selected on step
+		function successfulClick() {
+			parent[index].setAttribute("style", "display:none;");
```
Comment (NikolaiYurchenko) [inline layouts/partner-form/single.html]: cache parent[index] in variable

File: layouts/partner-form/single.html [hunk 2]
```diff
@@ -27,6 +27,10 @@
     var progressBarWrapper = document.getElementById("progress-bar-wrapper");
     var stepButtons = document.getElementById("step-buttons");
 
+		var useCaseOptions = document.getElementsByClassName('use-case');
```
Comment (NikolaiYurchenko) [inline layouts/partner-form/single.html]: indentations are broken time to time

File: layouts/partner-form/single.html [hunk 3]
```diff
@@ -54,23 +79,17 @@
...
-            parent[index].setAttribute("style", "display:block;");
-            if(parent[index].classList.contains('partner-form-per-month')) {
-              nextButton.setAttribute("style", "display:none;");
-              submitButton.setAttribute("style", "display:inline-block;");
-            }
+          } else if(parent[index].classList.contains("partner-form-use-case")) {
+						var selected = false;
+						for (var i = 0, length = useCaseOptions.length; i < length; i++) {
+		          if (useCaseOptions[i].selected) {
+								selected = true
```
Comment (NikolaiYurchenko) [inline layouts/partner-form/single.html]: semicolons are missing

File: layouts/partner-form/single.html [hunk 4]
```diff
@@ -54,23 +79,17 @@
...
+								successfulClick()
+		          }
+		        }
+						!selected && errElm[index].setAttribute("style", "display:block;");
```
Comment (NikolaiYurchenko) [inline layouts/partner-form/single.html]: IMO using if-statement is better for readability here

File: layouts/partner-form/single.html [hunk 5]
```diff
@@ -54,23 +79,17 @@
...
+					} else {
+            successfulClick()
```
Comment (NikolaiYurchenko) [inline layouts/partner-form/single.html]: missing semicolon

File: layouts/partner-form/single.html [hunk 6]
```diff
@@ -43,6 +47,27 @@
       }
     }
 
+		// if vale is input or selected on step
```
Comment (NikolaiYurchenko) [inline layouts/partner-form/single.html]: typo

File: layouts/partner-form/single.html [hunk 7]
```diff
@@ -123,43 +140,28 @@
         }
       }
 
-      var dataAmountOptions = document.getElementsByClassName('data-amount');
+			// handle steps with multiple options, one choice
+			function handleMultipleOptions(element, destination){
+				if(element) {
```
Comment (crawter) [inline layouts/partner-form/single.html]: little if is better 
if(!element) return;

...do some work

File: layouts/partner-form/single.html [hunk 8]
```diff
@@ -54,23 +79,17 @@
...
-              progressBar.setAttribute("style", `width: ${progressBarValue}%`);
-            }
-            index = index + 1
-            parent[index].setAttribute("style", "display:block;");
-            if(parent[index].classList.contains('partner-form-per-month')) {
-              nextButton.setAttribute("style", "display:none;");
-              submitButton.setAttribute("style", "display:inline-block;");
-            }
+          } else if(parent[index].classList.contains("partner-form-use-case")) {
+						var selected = false;
```
Comment (crawter) [inline layouts/partner-form/single.html]: let

File: layouts/partner-form/single.html [hunk 9]
```diff
@@ -54,23 +79,17 @@
...
+						for (var i = 0, length = useCaseOptions.length; i < length; i++) {
```
Comment (crawter) [inline layouts/partner-form/single.html]: let

File: layouts/partner-form/single.html [hunk 10]
```diff
@@ -54,23 +79,17 @@
...
+		          if (useCaseOptions[i].selected) {
+								selected = true
+								successfulClick()
+		          }
+		        }
+						!selected && errElm[index].setAttribute("style", "display:block;");
+					} else {
```
Comment (crawter) [inline layouts/partner-form/single.html]: do we need this else?

or we could call successfulClick without else?

File: layouts/partner-form/single.html [hunk 11]
```diff
@@ -43,6 +47,27 @@
...
+				progressBar.setAttribute("style", `width: ${progressBarValue}%`);
+			}
+			index = index + 1
+			parent[index].setAttribute("style", "display:block;");
+			if(parent[index].classList.contains('partner-form-per-month')) {
+				nextButton.setAttribute("style", "display:none;");
+				submitButton.setAttribute("style", "display:inline-block;");
+			}
+			errElm[index].setAttribute("style", "display:none");
+
```
Comment (crawter) [inline layouts/partner-form/single.html:111]: redundant empty line

File: layouts/partials/partner-form/form.html [hunk 12]
```diff
@@ -24,9 +24,9 @@ <h2>Great! Let's get started.</h2>
     </div>
 
     <div style="display:none" class="partner-form-last-name step active">
-      <label class="label">Your Last Name</label>
+      <label class="label">Your Last Name*</label>
```
Comment (crawter) [inline layouts/partials/partner-form/form.html]: why not bind label with input?

Comment (mbouzi) [reply layouts/partials/partner-form/form.html]: Sorry I'm not sure what you mean by this, if this helps for this section I was copying the visual styling of the imbedded form we were using previously

File: layouts/partner-form/single.html (hunk 1 above)
Comment (mbouzi) [reply layouts/partner-form/single.html]: putting this into a variable causes the code not to function properly

File: layouts/partner-form/single.html (hunk 10 above)
Comment (mbouzi) [reply layouts/partner-form/single.html]: I think the else is needed because it should not be called in certain scenarios

**Comment by mbouzi** [comment]
If the linting process still needs work, let me know and we can set up a call to go over!

File: package.json [hunk 13]
```diff
@@ -4,14 +4,20 @@
   "description": "Live at https://tardigrade.io",
   "main": "index.js",
   "dependencies": {
-    "eslint": "^5.16.0",
     "eslint-config-prettier": "^4.1.0",
```
Comment (crawter) [inline package.json]: i think all this dependencies could be places in dev dependencies

File: layouts/partner-form/single.html [hunk 14]
```diff
@@ -27,11 +27,15 @@
     var progressBarWrapper = document.getElementById("progress-bar-wrapper");
     var stepButtons = document.getElementById("step-buttons");
 
+		var useCaseOptions = document.getElementsByClassName('use-case');
+		var dataAmountOptions = document.getElementsByClassName('data-amount');
+		var fileSizeOptions = document.getElementsByClassName('file-size');
+		var perMonthOptions = document.getElementsByClassName('per-month');
```
Comment (crawter) [inline layouts/partner-form/single.html]: still broken indentations

File: package.json (hunk 13 above)
Comment (NikolaiYurchenko) [inline package.json]: move dev dependencies to "devDependencies"

File: package.json [hunk 15]
```diff
@@ -4,14 +4,20 @@
...
     "eslint-plugin-html": "^5.0.3",
     "eslint-plugin-prettier": "^3.0.1",
     "prettier": "^1.17.0",
     "prettier-eslint-cli": "^4.7.1"
   },
-  "devDependencies": {},
+  "devDependencies": {
+    "eslint": "^7.4.0",
```
Comment (NikolaiYurchenko) [inline package.json]: Remove ^ because it may cause incompatibilities in future

File: assets/scss/_404.scss [hunk 16]
```diff
@@ -1,16 +1,16 @@
 .p404-page {
   .p404 {
-		&__content-wrapper {
-			height: 350px;
-			@media (max-width: 991px) {
-	      height: 600px;
+    &__content-wrapper {
+      height: 350px;
```
Comment (NikolaiYurchenko) [inline assets/scss/_404.scss:4]: still broken indentations. Rules need to be added

File: layouts/partials/partner-form/form.html [hunk 17]
```diff
@@ -115,14 +115,14 @@ <h2>Great! Let's get started.</h2>
     </div>
 
     <div style="display:none" class="partner-form-how-long step active">
-      <label class="label">How long do you need the data to be stored on the network before you no longer need the data?</label>
+      <label class="label">How long do you need the data to be stored on the network before you no longer need the data?*</label>
```
Comment (crawter) [inline layouts/partials/partner-form/form.html]: it is a good practice to bind label with corresponding input (type text, email or password) with 'for' attribute
or smth like this:
`<label aria-label="mce-EMAIL-newsletter">`
`    <input class="subscribe-input" type="email" name="mce-EMAIL-newsletter" id="mce-EMAIL-newsletter" placeholder="Email Address" required="">`
`</label>`

File: layouts/partials/partner-form/form.html (hunk 12 above)
Comment (crawter) [reply layouts/partials/partner-form/form.html]: @mbouzi i have added more detailed comment here https://github.com/storj/tardigrade.io/pull/111#discussion_r450904514

File: layouts/partials/partner-form/form.html [hunk 18]
```diff
@@ -89,7 +89,7 @@ <h2>Great! Let's get started.</h2>
     </div>
 
     <div style="display:none" class="partner-form-data-amount  step active form-group">
-      <label class="label">WHat is the total amount of data that you will store in the Tardigrade platform (TB stored in a typical month)?</label>
+      <label class="label">WHat is the total amount of data that you will store in the Tardigrade platform (TB stored in a typical month)?*</label>
```
Comment (NikolaiYurchenko) [inline layouts/partials/partner-form/form.html]: typo "WHat"

File: layouts/partials/partner-form/form.html (hunk 17 above)
Comment (NikolaiYurchenko) [inline layouts/partials/partner-form/form.html]: add area-label attribute to label tag to bind it with inputs

**Comment by mbouzi** [comment]
@NikolaiYurchenko I'm sorry, I'm working on a bunch of things at once. Pushed the changes to the wrong branch

File: assets/scss/_404.scss (hunk 16 above)
Comment (mbouzi) [reply assets/scss/_404.scss:4]: Think I'm missing something, it looks like indentations were fixed after I ran "npm run fix-style"

**Pull request merged**